File: 1290-convert-binary-number-in-a-linked-list-to-integer/1290-convert-binary-number-in-a-linked-list-to-integer.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int BinToDec(string s){
        int n = s.length(),i=0;
        int j = n-1,sum=0;
        while(i<n){
            if(s[i]=='1')
                sum += pow(2,j);
            i++;j--;
        }
       return sum; 
    }
    
    int getDecimalValue(ListNode* head) {
        string s = "";
        while(head){
            s += to_string(head->val);
            head = head->next;
        }
        return BinToDec(s);
    }
};

File: 1290-convert-binary-number-in-a-linked-list-to-integer/1290-convert-binary-number-in-a-linked-list-to-integer.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def BinToDec(self,s):
        n = len(s)
        i,j,summ = 0,n-1,0
        while(i<n):
            if(s[i]=='1'):
                summ += 2**j
            i += 1
            j -= 1
        return summ
    
    def getDecimalValue(self, head: ListNode) -> int:
        s = ""
        while(head):
            s += str(head.val)
            head = head.next;
        return self.BinToDec(s)